src/system/behavior.cpp

```diff
@@ -267,6 +267,12 @@ void button_hold_callback(const uint8_t consecutiveButtonCheck, const uint32_t b
     realStartTime -= holdDuration;
   }
 
+  // prevent the '#38: luminosity ramp flashes' bug
+  if (isEndOfHoldEvent)
+  {
+    update_brightness(BRIGHTNESS, true);
+  }
+
```

Comment:
There's a cleaner way to handle this:
```diff
-  // prevent the '#38: luminosity ramp flashes' bug
-  if (isEndOfHoldEvent)
-  {
-    update_brightness(BRIGHTNESS, true);
-  }
```
and
```diff

  if (realStartTime < EARLY_ACTIONS_LIMIT_MS)
  {
-    if (isEndOfHoldEvent)
-      return;
```
with
```diff
    // 3+hold (3s): turn it on, with button usermode enabled
-    if (consecutiveButtonCheck == 3)
+    if (!isEndOfHoldEvent && consecutiveButtonCheck == 3)
    {
```
and
```diff
    // 4+hold (3s): turn it on, with bluetooth advertising
#ifdef USE_BLUETOOTH
-    if (consecutiveButtonCheck == 4)
+    if (!isEndOfHoldEvent && consecutiveButtonCheck == 4)
    {
```

The initial reason to cancel the event on `isEndOfHoldEvent` was to factorize code, however initially I didn't planned to let the brightness callbacks fall through. However, after initial testing, this was bothering me, thus I let the brightness calls went through, and forgot about removing the extra `if (isEndOfHoldEvent)`

I would prefer the way suggested above, as I find no point into duplicating the brightness update, while I think the suggestion above make things clearer (while being equivalent)